src/shared/module-util.c

```diff
@@ -50,22 +50,44 @@ int module_load_and_warn(struct kmod_ctx *ctx, const char *module, bool verbose)
                                          "Inserted module '%s'", kmod_module_get_name(mod));
                         else if (err == KMOD_PROBE_APPLY_BLACKLIST)
                                 log_full(verbose ? LOG_INFO : LOG_DEBUG,
-                                         "Module '%s' is deny-listed", kmod_module_get_name(mod));
-                        else {
-                                assert(err < 0);
-
-                                log_full_errno(!verbose ? LOG_DEBUG :
-                                               err == -ENODEV ? LOG_NOTICE :
-                                               err == -ENOENT ? LOG_WARNING :
-                                                                LOG_ERR,
-                                               err,
-                                               "Failed to insert module '%s': %m",
-                                               kmod_module_get_name(mod));
-                                if (!IN_SET(err, -ENODEV, -ENOENT))
-                                        r = err;
+                                         "Module '%s' is deny-listed (by kmod)", kmod_module_get_name(mod));
+                        else if (err == -EPERM) {
+                                STRV_FOREACH(i, blacklist) {
+                                        _cleanup_(kmod_module_unrefp) struct kmod_module *mod_blacklisted = NULL;
+
+                                        r = kmod_module_new_from_name(ctx, *i, &mod_blacklisted);
+                                        if (r < 0)
+                                                continue;
```

Comment:
This seems not necessary. The kernel command line option takes exact module names. Hence, simply compare the module name with each entry in the strv.

Reply:
Is there a strv function or so I can use to normalize module names like https://github.com/kmod-project/kmod/blob/master/shared/util.c#L151, i.e. replace all `-` with `_`?

Reply:
I do not think it is necessary to replace `-` with `_`. The kernel seem to accept exact module names: see https://github.com/torvalds/linux/blob/e8bc52cb8df80c31c73c726ab58ea9746e9ff734/kernel/module/main.c#L2258-L2277

Reply:
I ran some local tests and noticed that too.